Website/Providers/DataProviders/SqlDataProvider/07.04.01.SqlDataProvider

```diff
@@ -117,6 +117,49 @@ AS
 
 GO
 
+-- DNN-6702: fix an issue with shared modules in Site Groups
+-- ---------------------------------------------------------
+IF EXISTS(SELECT * FROM sys.indexes WHERE name = 'IX_{objectQualifier}ModulePermission' AND object_id = OBJECT_ID('{objectQualifier}ModulePermission'))
+	ALTER TABLE	{databaseOwner}[{objectQualifier}ModulePermission] DROP CONSTRAINT [IX_{objectQualifier}ModulePermission] 
+GO
+
+IF EXISTS(SELECT * FROM sys.indexes WHERE name = 'IX_{objectQualifier}ModulePermission_Portals' AND object_id = OBJECT_ID('{objectQualifier}ModulePermission'))
+	DROP INDEX [IX_{objectQualifier}ModulePermission_Portals] ON {databaseOwner}[{objectQualifier}ModulePermission]
+GO
+
+IF EXISTS(SELECT * FROM sys.indexes WHERE name = 'IX_{objectQualifier}ModulePermission_Modules' AND object_id = OBJECT_ID('{objectQualifier}ModulePermission'))
+	DROP INDEX [IX_{objectQualifier}ModulePermission_Modules] ON {databaseOwner}[{objectQualifier}ModulePermission]
+GO
+
+IF EXISTS(SELECT * FROM sys.indexes WHERE name = 'IX_{objectQualifier}ModulePermission_Users' AND object_id = OBJECT_ID('{objectQualifier}ModulePermission'))
+	DROP INDEX [IX_{objectQualifier}ModulePermission_Users] ON {databaseOwner}[{objectQualifier}ModulePermission]
+GO
+
+IF EXISTS(SELECT * FROM sys.indexes WHERE name = 'IX_{objectQualifier}ModulePermission_Roles' AND object_id = OBJECT_ID('{objectQualifier}ModulePermission'))
+	DROP INDEX [IX_{objectQualifier}ModulePermission_Roles] ON {databaseOwner}[{objectQualifier}ModulePermission]
+GO
+
+CREATE UNIQUE NONCLUSTERED INDEX IX_{objectQualifier}ModulePermission_Portals ON {databaseOwner}[{objectQualifier}ModulePermission]
+	(PortalID, ModuleID, PermissionID, RoleID, UserID) 
+	INCLUDE (AllowAccess)
+GO
+
```

Comment:
In looking through code, I can't find a single stored procedure which would use this query.  Even ones that look like they might are only called from deprecated methods and even then the actual query plan doesn't show this index getting used.  

As a result, please remove the ModulePermissions_Portals index as it appears unnecessary and will just add extra overhead to maintain the index.